src/robusta/integrations/kubernetes/custom_models.py

```diff
@@ -390,8 +395,30 @@ def get_single_pod(self) -> RobustaPod:
             raise Exception(f"got more pods than expected for job: {pods}")
         return pods[0]
 
+    def create_job_owned_secret(self, secret: JobSecret):
+        """
+            This secret will be auto-deleted when the pod is Terminated
+        """
+        # Due to inconsistant GC in K8s the OwnerReference needs to be the pod and not the job (Found in azure)
+        job_pod = self.get_single_pod()
+        robusta_owner_reference = OwnerReference(apiVersion="v1",
```

Comment:
Why do we use the pod as owner reference and not the job?
If the pod crashes, the secret will be deleted, right?

Reply:
yes the secret will be deleted. 
in the comment above I explained that when the job is the owner the GC doesn't clean up the secret at least in azure.
I am unsure why this behavior happens in k8s but due to being short on time I put off investigating in depth.